core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleWriterSuite.scala

```diff
@@ -0,0 +1,104 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort
+
+import java.io.File
+
+import org.mockito.Mockito._
+import org.mockito.MockitoAnnotations
+import org.scalatest.BeforeAndAfterEach
+import org.apache.spark.{ShuffleDependency, _}
+import org.apache.spark.memory.MemoryTestingUtils
+import org.apache.spark.serializer.JavaSerializer
+import org.apache.spark.shuffle.{BaseShuffleHandle, IndexShuffleBlockResolver}
+import org.apache.spark.util.Utils
+
+class SortShuffleWriterSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private val shuffleId = 0
+  private val numMaps = 5
+  private var outputFile: File = _
+  private val conf: SparkConf = new SparkConf(loadDefaults = false)
+  private val sc = new SparkContext("local-cluster[1,1,1024]", "test", conf)
+  private var shuffleHandle: BaseShuffleHandle[Int, Int, Int] = _
+  private val shuffleBlockResolver = new IndexShuffleBlockResolver(conf)
+  private val serializer = new JavaSerializer(conf)
+  private val context = MemoryTestingUtils.fakeTaskContext(sc.env)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    MockitoAnnotations.initMocks(this)
+    val partitioner = new Partitioner() {
+      def numPartitions = numMaps
+      def getPartition(key: Any) = Utils.nonNegativeMod(key.hashCode, numPartitions)
+    }
+    shuffleHandle = {
+      val dependency = mock(classOf[ShuffleDependency[Int, Int, Int]])
+      when(dependency.partitioner).thenReturn(partitioner)
+      when(dependency.serializer).thenReturn(serializer)
+      when(dependency.aggregator).thenReturn(None)
+      when(dependency.keyOrdering).thenReturn(None)
+      new BaseShuffleHandle(shuffleId, numMaps = numMaps, dependency)
+    }
+  }
+
+  override def afterEach(): Unit = {
+    try {
+      if (outputFile != null) {
+        Utils.deleteRecursively(outputFile)
+      }
+    } finally {
+      super.afterEach()
+    }
+  }
+
+  test("write empty iterator") {
+    val writer = new SortShuffleWriter[Int, Int, Int](
+      shuffleBlockResolver,
+      shuffleHandle,
+      mapId = 1, // MapId
+      context
+    )
+    writer.write(Iterator.empty)
+    writer.stop( /* success = */ true)
+    val dataFile = shuffleBlockResolver.getDataFile(shuffleId, 1)
+    assert(!dataFile.exists())
+    assert(dataFile.length() == 0)
+    assert(context.taskMetrics().shuffleWriteMetrics.bytesWritten == 0)
+    assert(context.taskMetrics().shuffleWriteMetrics.recordsWritten == 0)
+
+  }
+
+  test("write with some records") {
+    val records = Iterator((1, 2), (2, 3), (4, 4), (6, 5))
+    val writer = new SortShuffleWriter[Int, Int, Int](
+      shuffleBlockResolver,
+      shuffleHandle,
+      mapId = 2, // MapId
+      context
+    )
+    writer.write(records)
+    writer.stop( /* success = */ true)
+    val dataFile = shuffleBlockResolver.getDataFile(shuffleId, 2)
+    assert(dataFile.exists())
+    assert(dataFile.length() != 0)
```

Comment:
Can we compare the certain expected value instead of `!= 0` in these three lines?